vshard/replicaset.lua

```diff
@@ -225,6 +231,13 @@ local function replicaset_connect_to_replica(replicaset, replica)
         conn:on_connect(netbox_on_connect)
         conn.on_disconnect_ref = netbox_on_disconnect
         conn:on_disconnect(netbox_on_disconnect)
+        -- Whether name validation is needed.
+        conn.name = replica.id == replica.name
```

Comment:
This is a cool idea - getting the greeting and validating it on first access. We can piggyback here many other things, like vshard version, vshard features, etc. Lets make it more generic and introduce `conn.vconnect` - it would be a table storing the async requests for various stuff such as the name.

It would store members `is_named` (flags should have `is_` suffix or similar) and the other things that you need. Later we can add here more features, subscriptions, etc.

Reply:
Is it ok to use `storage_service_info`, like it's done now or do we want to introduce new call, specific for `conn.vconnect`? This means schema version bump, so I'd rather stay with `storage_service_info`.

Reply:
No, this is fine. And it wouldn't cause the version bump - `storage_service_info` works via `vshard.storage._call` which already exists and was created specifically for cases when we don't want to bother with schema changes but do want to change functions.

Reply:
> And it wouldn't cause the version bump

My bad.

> Lets make it more generic and introduce conn.vconnect

Done